kratos/utilities/volume_inside_voxel_utility.cpp

```diff
@@ -0,0 +1,382 @@
+        GeometryPtrType pGeom =Kratos::make_shared<Quadrilateral3D4<NodeType>>(rPoints);
+        return pGeom->Volume();
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    double VolumeInsideVoxelUtility::EdgeFilledPortion(std::vector<double>& Distances, const PointsArrayType& rEnds) {
+        const double Length = Distances[Distances.size() - 1];
+        double portion = 0; 
+        bool inside;       
+
+        //casuistics: Both nodes inside, both nodes outside, one node inside and one outside
+        if (rEnds[0].GetSolutionStepValue(DISTANCE) > 0 && rEnds[1].GetSolutionStepValue(DISTANCE) > 0) {
+            bool inside = true;
+            if(Distances.size() % 2 == 1) Distances.pop_back();
+            if (Distances.size() == 2) return 1;
+        } else if (rEnds[0].GetSolutionStepValue(DISTANCE) > 0 && rEnds[1].GetSolutionStepValue(DISTANCE) < 0) {
+            if(Distances.size() % 2 == 0) Distances.pop_back();
+            inside = true;
+        } else if (rEnds[0].GetSolutionStepValue(DISTANCE) < 0 && rEnds[1].GetSolutionStepValue(DISTANCE) > 0) {
+            if(Distances.size() % 2 == 0) Distances.pop_back();
+            inside = false;
+        } else {    //rEnds[0].GetSolutionStepValue(DISTANCE) < 0 && rEnds[1].GetSolutionStepValue(DISTANCE) < 0
+            if (Distances.size() % 2 == 1) Distances.pop_back();
+            if (Distances.size() == 2) return 0;
+            inside = false;
+        }
+
+        for(int i = 1; i < Distances.size(); i++) {
+            if (inside) {
+                portion += abs(Distances[i]-Distances[i-1])/Length;
+                inside = false;
+            } else inside = true;             
+        }
+        return portion;
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    double VolumeInsideVoxelUtility::GetFactor(const PointsArrayType& nodes, const std::vector<std::vector<double>>& neighbours,const int node) {
+
+        if( nodes[node].GetSolutionStepValue(DISTANCE) > 0 && nodes[neighbours[node][0]].GetSolutionStepValue(DISTANCE) < 0 &&
+            nodes[neighbours[node][1]].GetSolutionStepValue(DISTANCE) < 0 || nodes[node].GetSolutionStepValue(DISTANCE) < 0 && 
+            nodes[neighbours[node][0]].GetSolutionStepValue(DISTANCE) > 0 && nodes[neighbours[node][1]].GetSolutionStepValue(DISTANCE) > 0) {
+                return 0.5;
+            }
+        return 1;
+    }
+
+    /***********************************************************************************
+     **********************************************************************************/
+
+    int VolumeInsideVoxelUtility::GetCase(const PointsArrayType& nodes, const std::vector<std::vector<double>>& neighbours,const int node) {
```

Comment:
```suggestion
    int VolumeInsideVoxelUtility::GetCase(
        const PointsArrayType& rNodes,
        const std::vector<std::vector<double>>& rNeighbours,
        const int NodeIndex)
    {
```
Please stick to the style guide. Also note that by Node we usually refer to the object, you intend to pass the node index.